### ISSUE TITLE
[FEATURE] Support passing few-shot examples to Neo4jContentRetriever

### ISSUE THREAD

**Issue by sjivan** [langchain4j/langchain4j#1319]
Andreas Kollegger course “Knowledge Graphs for RAG” covers creating a Neo4J Knowledge Graph and in his example he illustrates show without questions and cypher examples, the Neo4jContentRetriever gets the wrong Cypher.

This is also covered in the "Final Part" section of the blog below which is essentially the same as  Andreas' Kollegger course 
https://medium.com/@rubenszimbres/building-knowledge-graphs-from-scratch-using-neo4j-and-vertex-ai-8311eb69a472

I'd like Neo4jContentRetriever builder to support passing in an "examples" string as below

```java
Neo4jContentRetriever retriever = Neo4jContentRetriever.builder()
                .graph(graph)
                .chatLanguageModel(chatLanguageModel)
                .examples(examples)
                .build();
```

with the default prompt template being

```java
private static final PromptTemplate DEFAULT_PROMPT_TEMPLATE = PromptTemplate.from("""

            Task:Generate Cypher statement to query a graph database.
            Instructions
            Use only the provided relationship types and properties in the schema.
            Do not use any other relationship types or properties that are not provided.
            Schema:
            {{schema}}
            
            Do not respond to any questions that might ask anything else than for you to construct a Cypher statement.
           
            {{examples}}
            
            Question: {{question}}
            
            """);
```

and pass "examples" variable as "" if not set by the user. Apologies for the formatting, couldn't get it to display right.

**Comment by langchain4j** [comment]
@sjivan this can be achieved by providing a `PromptTemplate` in the builder of `Neo4jContentRetriever`:
```java
Neo4jContentRetriever.builder()
                .graph(graph)
                .chatLanguageModel(chatLanguageModel)
                .promptTemplate(PromptTemplate.from(...))
                .build();
```

**Comment by sjivan** [comment]
> @sjivan this can be achieved by providing a `PromptTemplate` in the builder of `Neo4jContentRetriever`:
> 
> ```java
> Neo4jContentRetriever.builder()
>                 .graph(graph)
>                 .chatLanguageModel(chatLanguageModel)
>                 .promptTemplate(PromptTemplate.from(...))
>                 .build();
> ```

@langchain4j Agree but the base template is a good one that I just need to add examples to. If I were to pass a promptTemplate it would essentially be the default template plus the examples. After going through Andreas Kollegger's course it's quite evident that one almost always needs to pass examples of Cypher query for the Neo4jContentRetriever to return working Cypher so it feels that it's best to support passing "examples" as a first class construct.

**Comment by langchain4j** [comment]
Ok, makes sense. But please don't make examples mandatory.

**Comment by jexp** [comment]
Yes makes sense to add an optional examples, section in the prompt, not sure how the other retrievers handle this in langchain4j?

**Comment by langchain4j** [comment]
There are usually no examples. But having examples is a good idea, I would just keep them optional and not force the user to provide them

**Comment by sjivan** [comment]
@jexp  @vga91 any thoughts on moving this to the ommunity repo as suggested by @dliubarskyi ?
